cmd/cmd.go

```diff
@@ -234,24 +236,36 @@ func (o *options) sternConfig() (*stern.Config, error) {
 		return nil, errors.New("color should be one of 'always', 'never', or 'auto'")
 	}
 
+	messageTpl := "{{.Message}}"
+	switch o.inputFormat {
+	case "zap-json":
+		if color.NoColor {
+			messageTpl = `{{ with $msg := .Message | parseJSON }}[{{ $msg.ts }}] {{ $msg.level }} ({{ $msg.caller }}) {{ $msg.msg }}{{ end }}`
+		} else {
+			messageTpl = `{{ with $msg := .Message | parseJSON }}[{{ color $.TimestampColor $msg.ts }}] {{ levelColor $msg.level }} ({{ color $.CallerColor $msg.caller }}) {{ $msg.msg }}{{ end }}`
```

Comment:
I would like to create a generic color function and use it as follows. This will eliminate the need to change `tail.go`.

```diff
diff --git a/cmd/cmd.go b/cmd/cmd.go
index 5dc6784..4395441 100644
--- a/cmd/cmd.go
+++ b/cmd/cmd.go
@@ -242,7 +242,7 @@ func (o *options) sternConfig() (*stern.Config, error) {
                if color.NoColor {
                        messageTpl = `{{ with $msg := .Message | parseJSON }}[{{ $msg.ts }}] {{ $msg.level }} ({{ $msg.caller }}) {{ $msg.msg }}{{ end }}`
                } else {
-                       messageTpl = `{{ with $msg := .Message | parseJSON }}[{{ color $.TimestampColor $msg.ts }}] {{ levelColor $msg.level }} ({{ color $.CallerColor $msg.caller }}) {{ $msg.msg }}{{ end }}`
+                       messageTpl = `{{ with $msg := .Message | parseJSON }}[{{ colorGreen $msg.ts }}] {{ levelColor $msg.level }} ({{ colorCyan $msg.caller }}) {{ $msg.msg }}{{ end }}`
                }
        case "default":
        default:
@@ -312,6 +312,14 @@ func (o *options) sternConfig() (*stern.Config, error) {
                "color": func(color color.Color, text string) string {
                        return color.SprintFunc()(text)
                },
+               "colorBlack":   color.BlackString,
+               "colorRed":     color.RedString,
+               "colorGreen":   color.GreenString,
+               "colorYellow":  color.YellowString,
+               "colorBlue":    color.BlueString,
+               "colorMagenta": color.MagentaString,
+               "colorCyan":    color.CyanString,
+               "colorWhite":   color.WhiteString,
                "levelColor": func(level string) string {
                        var levelColor *color.Color
                        switch strings.ToLower(level) {
```

Reply:
Thank you for the suggestion! I was not happy about adding specific color fields into the `tail.go` file, and I was thinking more about function like `colorFromString`, or even change default `color` function to test agains specific constants, but I was not sure whether it worth the trouble.
I like your idea, will apply your suggestion 👍